extension/extensionlimiter/limiterhelper/wrapper.go

```diff
@@ -0,0 +1,125 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package limiterhelper // import "go.opentelemetry.io/collector/extension/extensionlimiter/limiterhelper"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/extension/extensionlimiter"
+)
+
+// LimiterWrapperProvider provides access to LimiterWrappers, which is
+// the appropriate interface for callers that can easily wrap a
+// function call, because for wrapped calls there is no distinction
+// between rate limiters and resource limiters.
+type LimiterWrapperProvider interface {
```

Comment:
Do you expect others to implement this interface? Otherwise we should have this as a struct.

Reply:
I see. I do not need this to be an interface, however I wonder about all the mock-based tests I've ever written with `gomock` and so on, whether we lose access to this style of test. I'll make this a struct.